src/commands/predict.rs

```diff
@@ -0,0 +1,96 @@
+use serenity::all::{
+    CommandDataOptionValue, CommandOptionType
+};
+use serenity::builder::{
+    CreateCommand, CreateCommandOption, CreateEmbed,
+    CreateInteractionResponseMessage, CreateEmbedAuthor,
+    CreateEmbedFooter,
+};
+use serenity::client::Context;
+use serenity::model::application::CommandInteraction;
+use serenity::model::Color;
+
+use crate::api::vrc_data_analysis::VRCDataAnalysis;
+
+#[derive(Default, Clone, Debug, PartialEq)]
+pub struct PredictCommand;
+
+impl PredictCommand {
+    pub fn command() -> CreateCommand {
+        CreateCommand::new("predict")
+			.description("Predict the outcome of a VRC match. Use \"AVG\" to represent an average team.")
+			.add_option(
+				CreateCommandOption::new(CommandOptionType::String, "r1", "Red Alliance Partner 1")
+                    .required(true)
+			)
+            .add_option(
+				CreateCommandOption::new(CommandOptionType::String, "r2", "Red Alliance Partner 2")
+                    .required(true)
+			)
+            .add_option(
+				CreateCommandOption::new(CommandOptionType::String, "b1", "Blue Alliance Partner 1")
+                    .required(true)
+			)
+            .add_option(
+				CreateCommandOption::new(CommandOptionType::String, "b2", "Blue Alliance Partner 2")
+                    .required(true)
+			)
+    }
+
+    fn progress_bar(length: usize, progress: f64) -> String {
+        let red_dot_count = (length as f64 * (progress / 100.0)).round() as usize;
+
+        "🟥".repeat(red_dot_count) + "🟦".repeat(length - red_dot_count).as_ref()
+    }
+
+    pub async fn response(
+        &self,
+        _ctx: &Context,
+        interaction: &CommandInteraction,
+        vrc_data_analysis: &VRCDataAnalysis
+    ) -> CreateInteractionResponseMessage {
+        let r1 = if let CommandDataOptionValue::String(number) = &interaction.data.options[0].value {
```

Comment:
There has got to be a better way to do this lol.

Reply:
indeed